common/test/acceptance/tests/lms/test_lms_cohorted_courseware_search.py

```diff
@@ -0,0 +1,244 @@
+
+        group_configurations_page.create_first_content_group()
+        config = group_configurations_page.content_groups[0]
+        config.name = self.content_group_a
+        config.save()
+
+        group_configurations_page.add_content_group()
+        config = group_configurations_page.content_groups[1]
+        config.name = self.content_group_b
+        config.save()
+
+    def link_html_to_content_groups_and_publish(self):
+        """
+        Updates 3 of the 4 existing html to limit their visibility by content group.
+        Publishes the modified units.
+        """
+        container_page = self.go_to_unit_page()
+
+        def set_visibility(html_block_index, content_group, second_content_group=None):
+            """
+            Set visibility on html blocks to specified groups.
+            """
+            html_block = container_page.xblocks[html_block_index]
+            html_block.edit_visibility()
+            if second_content_group:
+                ComponentVisibilityEditorView(self.browser, html_block.locator).select_option(
+                    second_content_group, save=False
+                )
+            ComponentVisibilityEditorView(self.browser, html_block.locator).select_option(content_group)
+
+        set_visibility(1, self.content_group_a)
+        set_visibility(2, self.content_group_b)
+        set_visibility(3, self.content_group_a, self.content_group_b)
+
+        container_page.publish_action.click()
+
+    def create_cohorts_and_assign_students(self):
+        """
+        Adds 2 manual cohorts, linked to content groups, to the course.
+        Each cohort is assigned one student.
+        """
+        instructor_dashboard_page = InstructorDashboardPage(self.browser, self.course_id)
+        instructor_dashboard_page.visit()
+        cohort_management_page = instructor_dashboard_page.select_cohort_management()
+
+        def add_cohort_with_student(cohort_name, content_group, student):
+            """
+            Create cohort and assign student to it.
+            """
+            cohort_management_page.add_cohort(cohort_name, content_group=content_group)
+            # After adding the cohort, it should automatically be selected
+            EmptyPromise(
+                lambda: cohort_name == cohort_management_page.get_selected_cohort(), "Waiting for new cohort"
+            ).fulfill()
+            cohort_management_page.add_students_to_selected_cohort([student])
+        add_cohort_with_student("Cohort A", self.content_group_a, self.cohort_a_student_username)
+        add_cohort_with_student("Cohort B", self.content_group_b, self.cohort_b_student_username)
+        cohort_management_page.wait_for_ajax()
+
+    def test_page_existence(self):
+        """
+        Make sure that the page is accessible.
+        """
+        self._auto_auth(self.USERNAME, self.EMAIL, False)
+        self.courseware_search_page.visit()
+
+    def test_cohorted_search_user_a_a_content(self):
```

Comment:
Add a test to verify that staff users see all content by default. Then add a second test which verifies that they see only the appropriate content when previewing as belonging to a particular content group.